### ISSUE TITLE
Callable.callv() silently ignores call if parameters don't match

### ISSUE THREAD

**Issue by speveril** [godotengine/godot#69237]
### Godot version

4.0.beta6.official

### System information

Windows 11

### Issue description

Calling `.callv()` on a Callable and passing an array with an incorrect number of parameters will just silently ignore the call -- the Callable is not executed and there is no error reported in the log or in the debugger.

Note that `.call()` *does* generate a run time error if called with the wrong number of parameters.

### Steps to reproduce

```
func _ready():
    my_func.callv(["A", "B"])

func my_func(x:
    print(x)
```

will run without error, but nothing will be printed to the log.

The attached repro project duplicates the above code, essentially, with a few extra calls to show that the function works normally. 

### Minimal reproduction project

[bugrepro-callvfailure.zip](https://github.com/godotengine/godot/files/10097214/bugrepro-callvfailure.zip)

**Comment by rburing** [comment]
This is true, I didn't add error reporting in https://github.com/godotengine/godot/pull/66177. Since passing parameters that don't match the signature is a programmer error, I think checking the error code and printing an error should be done only in debug builds.

Related to https://github.com/godotengine/godot/issues/62838.

**Comment by freehuntx** [comment]
@rburing in my case i need dynamic method creation + calling over network. So methods get generated dynamically and can be called dynamically.
Right now i have to manage each function and store the expected argument count and check it myself.

I would love if callv behaves the same as call. (Not just in debug builds)
